### ISSUE TITLE
Undocumented change in node "engines" requirement from mapbox-gl@2.3.0→mapbox-gl@2.4.0

### ISSUE THREAD

**Issue by danvk** [mapbox/mapbox-gl-js#10914]
**mapbox-gl-js version**: 2.4.0

**browser**: n/a

### Steps to Trigger Behavior

```
$ nvm use 12.16.3
$ node --version
v12.16.3
$ yarn add mapbox-gl
yarn add v1.17.3
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
error mapbox-gl@2.4.0: The engine "node" is incompatible with this module. Expected version ">=14.15.4". Got "12.16.3"
error Found incompatible module.
```

### Link to Demonstration

(see above)

### Expected Behavior

mapbox-gl is a client-side library. It doesn't make sense for it to require a specific node version.

I understand that the mapbox-gl team requires Node 14+ for development, but I don't understand why that requirement needs to be imposed on Mapbox's users as well.

This seems to be an [undocumented](https://github.com/mapbox/mapbox-gl-js/releases/tag/v2.4.0) breaking change in going from Mapbox GL JS 2.3.0→2.4.0.

### Actual Behavior

mapbox-gl should work on Node 12. I'm not sure it makes sense to include "engines" at all in the `package.json` that's distributed on npm, but the old requirement (Node 6+) was so old that it's unlikely anyone noticed. See https://github.com/mapbox/mapbox-gl-js/issues/8528

**Comment by rreusser** [comment]
Thanks for bringing this up, @danvk. This seems like an unintended consequence of this PR: https://github.com/mapbox/mapbox-gl-js/pull/10792 which addressed this issue: https://github.com/mapbox/mapbox-gl-js/issues/10778

**Comment by rreusser** [comment]
It appears that `npm install mapbox-gl` succeeds as well as `yarn add mapbox-gl --ignore-engines`, after which `yarn install` succeeds. I think you're correct though that this doesn't seem correct to impose upon users who aren't even using it in a `node` environment.

**Comment by ryanhamley** [comment]
I think maybe we need to revert the change in `package.json` and instead update `style-spec/package.json` with this change

**Comment by mourner** [comment]
@ryanhamley considering that `style-spec` provides a CJS bundle (through the `main` entry point) that works fine on older Node versions, I would suggest not using the `engines` field there too.

**Comment by mourner** [comment]
Fixed in v2.4.1.
